classes/fetch_services.py

```diff
@@ -124,16 +128,47 @@ def parse(self, response) -> dict:
         upload_date = datetime.strptime(response.get("upload_date"), date_format)
         upload_date = upload_date.replace(tzinfo=timezone.utc)
 
+        if response.get("duration") is None and prompt_dur:
```

Comment:
Using `prompt_dur` here may cause other applications which rely on `YtDlpFetchService` to break. The reason is that `prompt_dur` is a global dependency; in order to ensure that it's set, `prompt_for_missing_durs` has to be run first, so any code that currently isn't doing that will likely break due to not knowing what `prompt_dur` is.

Passing the `prompt_dur` dependency as an argument to `__init__`, as suggested above, would fix this problem, since it would be possible to provide a default value for any code that isn't currently setting it.